fili-core/src/main/java/com/yahoo/bard/webservice/web/apirequest/generator/metric/ProtocolLogicalMetricGenerator.java

```diff
@@ -135,25 +109,37 @@ public void validate(
      * prefer using a generator instance instead.
      *
      * @param apiMetricQuery  URL query string containing the metrics separated by ','
+     * @param requestGranularity Granularity of the request
      * @param metricDictionary  Metric dictionary contains the map of valid metric names and logical metric objects
      *
      * @return set of metric objects
      */
+    @Override
     public LinkedHashSet<LogicalMetric> generateLogicalMetrics(
             String apiMetricQuery,
+            Granularity requestGranularity,
             MetricDictionary metricDictionary
     ) {
-        List<ApiMetric> apiMetrics = parseApiMetricQuery(apiMetricQuery);
+        List<ApiMetric> apiMetrics = parseApiMetricQueryWithGranularity(apiMetricQuery, requestGranularity);
```

Comment:
We can probably drop the custom name and rely on the method signature to distinguish them.